api/src/main/java/org/openmrs/module/fhir2/providers/r3/LocationFhirResourceProvider.java

```diff
@@ -63,6 +72,35 @@ public Location getLocationById(@IdParam @NotNull IdType id) {
 		return Location30_40.convertLocation(location);
 	}
 	
+	@Create
+	public MethodOutcome createLocation(@ResourceParam Location location) {
+		return FhirProviderUtils.buildCreate(
+			    locationService.create(Location30_40.convertLocation(location)));
+	} 
+
+	@Update
+	@SuppressWarnings("unused")
+	public MethodOutcome updateLocation(@IdParam IdType id, @ResourceParam Location location) {
+		if (id == null || id.getIdPart() == null) {
+			throw new InvalidRequestException("id must be specified to update");
+		}
+
+		location.setId(id.getIdPart());
+
+		return FhirProviderUtils.buildUpdate(locationService.update(id.getIdPart(),
+		    Location30_40.convertLocation(location)));
+	}
+
+	@Delete
+	@SuppressWarnings("unused")
+	public OperationOutcome deleteLocation(@IdParam @NotNull IdType id) {
+		org.hl7.fhir.r4.model.Location location = locationService.delete(id.getIdPart());
+		if (location == null) {
+			throw new ResourceNotFoundException("Could not find location to delete with id " + id.getIdPart());
+		}
+		return OperationOutcome30_40.convertOperationOutcome(FhirProviderUtils.buildDelete(location));
```

Comment:
Please rewrite this as `FhirProviderUtils.buildDelete(Location30_40.convertLocation(location))`